### PR TITLE
Add text input

### PR DESCRIPTION
**Pull request by mfrachet**
# TextInput

## Description

Higher level component for displaying a textinput on the screen.

NOTE: the Link component is not part of the API since it does not exist yet in the DS document

## Demo
Example on : https://design-system-git-add-text-input-strapijs.vercel.app/?path=/story/textinput--base

## API

```jsx
  <TextInput
          placeholder="This is a content placeholder"
          label="Content"
          name="content"
          hint="Description line"
          error={undefined}
          onChange={(e) => {}}
          value={'content'}
          action={<span>Hello world</span>}
        />
```

### CODE REVIEW

File: packages/strapi-design-system/src/TextInput/TextInput.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+export const TextInput = React.forwardRef(
+  ({ leftAction, rightAction, name, hint, error, label, action, ...props }, ref) => {
+    return (
+      <TextareaWrapper>
+        <Field name={name} hint={hint} error={error}>
+          <Stack size={1}>
+            <Row cols="auto auto 1fr" gap={1}>
+              <FieldLabel>{label}</FieldLabel>
+              {action && <Box paddingLeft={1}>{action}</Box>}
```
Comment (HichamELBSI) [inline packages/strapi-design-system/src/TextInput/TextInput.js:23]: The difference between `leftAction, rightAction and action` props is not very clear for me to be honest. In which use cases should we use `leftAction/rightAction` vs `action`.

**Review by HichamELBSI — CHANGES_REQUESTED** [review]

File: packages/strapi-design-system/src/TextInput/TextInput.js (hunk 1 above)
Comment (mfrachet) [reply packages/strapi-design-system/src/TextInput/TextInput.js:23]: Oh yeah I understand 😊 . Left/ Right actions are for the input component (like the "eye" for the password, or the "magnifier" for the search input"). The action is for the tooltip information.

Do you have something in mind to adjust this?

Comment (HichamELBSI) [reply packages/strapi-design-system/src/TextInput/TextInput.js:23]: WDYT about `labelAction`

Comment (mfrachet) [reply packages/strapi-design-system/src/TextInput/TextInput.js:23]: Let's try with that 👍🏻

**Review by HichamELBSI — APPROVED** [review]
LGTM

**Pull request merged**